app/src/androidTest/java/ch/epfl/sdp/blindly/settings/SettingsTest.kt

```diff
@@ -102,34 +103,61 @@ class SettingsTest {
         onView(withId(R.id.done_button)).perform(click())
 
         assertEquals(RESULT_OK, activityRule.scenario.result.resultCode)
-        assertEquals(TEST_SHOW_ME_MEN,
+        assertEquals(
+            TEST_SHOW_ME_MEN,
             activityRule.scenario.result.resultData.getStringExtra(EXTRA_SHOW_ME)
         )
     }
 
-    private fun setProgress(progress: Int): ViewAction {
+    private fun setSliderValue(value: Int): ViewAction {
         return object : ViewAction {
             override fun perform(uiController: UiController?, view: View) {
-                val seekBar = view as SeekBar
-                seekBar.progress = progress
+                val slider = view as Slider
+                slider.value = value.toFloat()
             }
 
             override fun getDescription(): String {
-                return "Set a progress on a SeekBar"
+                return "Set a value on a Slider"
             }
 
             override fun getConstraints(): org.hamcrest.Matcher<View>? {
-                return ViewMatchers.isAssignableFrom(SeekBar::class.java)
+                return isAssignableFrom(Slider::class.java)
+            }
+        }
+    }
+
+    private fun setRangeSliderValues(minValue: Int, maxValue: Int): ViewAction {
+        return object : ViewAction {
+            override fun perform(uiController: UiController?, view: View) {
+                val rangeSlider = view as RangeSlider
+                rangeSlider.values = listOf(minValue.toFloat(), maxValue.toFloat())
+            }
+
+            override fun getDescription(): String {
+                return "Set the values on a RangeSlider"
+            }
+
+            override fun getConstraints(): org.hamcrest.Matcher<View>? {
+                return isAssignableFrom(RangeSlider::class.java)
             }
         }
     }
 
     @Test
-    fun movingTheSeekBarChangesTheRadiusText() {
+    fun movingTheSliderChangesTheRadiusText() {
         init()
-        onView(withId(R.id.seekBar)).perform(setProgress(80))
+        onView(withId(R.id.location_slider)).perform(setSliderValue(80))
         val radiusText = onView(withId(R.id.radius_text))
-        radiusText.check(ViewAssertions.matches(ViewMatchers.withText(TEST_RADIUS)))
+        radiusText.check(matches(withText(TEST_RADIUS)))
+        release()
+    }
+
+    @Test
+    fun movingTheRangeSliderChangesTheSelectedAgeRangeText() {
+        init()
+        onView(withId(R.id.age_range_slider)).perform(setRangeSliderValues(40, 60))

```

Comment:
Maybe you could replace the magic numbers 40 and 60 with constants

Reply:
good idea :)